nextcord/abc.py

```diff
@@ -86,6 +87,11 @@
         OverwriteType,
         PermissionOverwrite as PermissionOverwritePayload,
     )
+    from .types.components import Component as ComponentT
+    from .types.message import (
+        AllowedMentions as AllowedMentionsT,
+        MessageReference as MessageReferencePayloadT,
+    )
```

Comment:
The convention used elsewhere for importing types is to just add `Payload`

```suggestion
    from .types.components import Component as ComponentPayload
    from .types.message import (
        AllowedMentions as AllowedMentionsPayload,
        MessageReference as MessageReferencePayload,
    )
```